### PR TITLE
Add token limit

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Limit the amount of tokens that may be emitted during a lex.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/parser2/lexer/lexer.go [hunk 1]
```diff
@@ -27,6 +27,19 @@ import (
 	"github.com/onflow/cadence/runtime/common"
 )
 
+// tokenLimit is a sensible limit for how many tokens may be emitted
+const tokenLimit = 2 << 18
```
Comment (robert-e-davidson3) [inline runtime/parser2/lexer/lexer.go:31]: Why `2 << 18` instead of `1 << 19`?

File: runtime/parser2/lexer/lexer.go [hunk 2]
```diff
@@ -229,6 +242,11 @@ func (l *lexer) acceptOne(r rune) bool {
 
 // emit writes a token to the channel.
 func (l *lexer) emit(ty TokenType, val interface{}, rangeStart ast.Position, consume bool) {
+
+	if len(l.tokens) >= tokenLimit {
+		panic(TokenLimitReachedError{})
```
Comment (robert-e-davidson3) [inline runtime/parser2/lexer/lexer.go:247]: Should this be wrapped in `FatalError`? I'm not sure of the flow of control so I really don't know.

**Review by robert-e-davidson3 — APPROVED** [review]

File: runtime/parser2/lexer/lexer.go (hunk 2 above)
Comment (turbolent) [reply runtime/parser2/lexer/lexer.go:247]: Tokenization happens before parsing, so there is no real need to wrap it. We might want to clean up / unify the error handling in general

File: runtime/parser2/lexer/lexer.go (hunk 1 above)
Comment (turbolent) [inline runtime/parser2/lexer/lexer.go:31]:
```suggestion
const tokenLimit = 1 << 19
```

**Review by SupunS — APPROVED** [review]

**Pull request merged**